src/Symfony/Component/Cache/Messenger/Message.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Messenger;
+
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\DependencyInjection\ReverseContainer;
+
+/**
+ * Conveys a cached value that needs to be computed.
+ */
+final class Message
+{
+    private $item;
+    private $pool;
+    private $callback;
+
+    public static function create(ReverseContainer $reverseContainer, callable $callback, CacheItem $item, AdapterInterface $pool): ?self
+    {
+        try {
+            $item = clone $item;
+            $item->set(null);
+        } catch (\Exception $e) {
+            return null;
+        }
+
+        $pool = $reverseContainer->getId($pool);
+
+        if (\is_object($callback)) {
+            if (null === $id = $reverseContainer->getId($callback)) {
+                return null;
+            }
+
+            $callback = '@'.$id;
+        } elseif (!\is_array($callback)) {
+            $callback = (string) $callback;
+        } elseif (!\is_object($callback[0])) {
+            $callback = [(string) $callback[0], (string) $callback[1]];
+        } else {
+            if (null === $id = $reverseContainer->getId($callback[0])) {
+                return null;
+            }
+
+            $callback = ['@'.$id, (string) $callback[1]];
+        }
+
+        return new self($item, $pool, $callback);
+    }
+
+    /**
+     * @internal use Message::create() instead
+     */
+    public function __construct(CacheItem $item, string $pool, $callback)
```

Comment:
Why not make it private?

Reply:
At the origin: because the Serializer component was wired by default, and this was a requirement.
Now removed.